tests/python/pants_test/test_base.py

```diff
@@ -674,3 +675,57 @@ def captured_logging(self, level=None):
     finally:
       root_logger.setLevel(old_level)
       root_logger.removeHandler(handler)
+
+  def retrieve_single_product_at_target_base(self, product_mapping, target):
+    mapping_for_target = product_mapping.get(target)
+    single_base_dir = assert_single_element(list(mapping_for_target.keys()))
+    single_product = assert_single_element(mapping_for_target[single_base_dir])
+    return single_product
+
+  def populate_target_dict(self, target_map):
+    """Return a dict containing targets with files generated according to `target_map`.
+
+    The values of `target_map` should themselves be a flat dict, which contains keyword arguments
+    fed into `self.make_target()`, along with a few special keys. Special keys are:
+    - 'key' (required -- used to index into the returned dict).
```

Comment:
Could this be an address, or address string?

Reply:
It *could* default to that, actually -- the idea was to decouple accessing the target from within testing from its specific address so you could use a test class-specific shorthand.

Reply:
It now defaults to the target address string!